packages/react/src/components/Chat/ChatItem.tsx

```diff
@@ -89,23 +92,58 @@ class ChatItem extends UIComponent<ReactProps<ChatItemProps>, any> {
   }
 
   private renderChatItem(styles: ComponentSlotStylesPrepared) {
-    const { message, gutter, contentPosition } = this.props
+    const { gutter, contentPosition } = this.props
     const gutterElement =
       gutter &&
       Box.create(gutter, {
         defaultProps: { className: ChatItem.slotClassNames.gutter, styles: styles.gutter },
       })
 
+    const messageElement = this.setAttachedPropValueForChatMessage(styles)
+
     return (
       <>
         {contentPosition === 'start' && gutterElement}
-        {Box.create(message, {
-          defaultProps: { className: ChatItem.slotClassNames.message, styles: styles.message },
-        })}
+        {messageElement}
         {contentPosition === 'end' && gutterElement}
       </>
     )
   }
+
+  setAttachedPropValueForChatMessage = styles => {
+    const { message, attached } = this.props
+    const messageElement = Box.create(message, {
+      defaultProps: {
+        className: ChatItem.slotClassNames.message,
+        styles: styles.message,
+      },
+    })
+
+    // the element is ChatMessage
+    if (ChatMessage.isTypeOfElement(messageElement)) {
+      return this.cloneElementWithCustomProps(messageElement, { attached })
+    }
+
+    // the children is ChatMessage
+    if (ChatMessage.isTypeOfElement(getElementProp(messageElement, 'children'))) {
+      return this.cloneElementWithCustomProps(messageElement, { attached }, 'children')
+    }
+
+    // the content is ChatMessage
+    if (ChatMessage.isTypeOfElement(getElementProp(messageElement, 'content'))) {
+      return this.cloneElementWithCustomProps(messageElement, { attached }, 'content')
+    }
```

Comment:
Can you please clarify this for me? We will add `attached` prop only is `element`/`children`/`content` are `ChatItem`?
But, any HOC will break this: https://codesandbox.io/s/n4mm1ny55l

Reply:
Yes, by default we will propagate the attached in these three use cases (which we see as the most common) - the attached prop can always be applied on the ChatMessage if it is wrapped... I am open for refactoring it if we have some better ideas.. (once the template mechanism will be available to the user, we can safely omit these changes)